Utils/Extensions/StringExtensions.swift

```diff
@@ -40,4 +44,105 @@ extension String {
     public var asURL: NSURL? {
         return NSURL(string: self)
     }
+
+    private func shaw256(salt: String, payload: String) -> String? {
+        var saltData = salt.dataUsingEncoding(NSUTF8StringEncoding)
+        var payloadData = payload.dataUsingEncoding(NSUTF8StringEncoding)
+        let hash = NSMutableData(length: Int(CC_SHA256_DIGEST_LENGTH))
+
+        let operation: CCOperation = UInt32(kCCHmacAlgSHA256)
+        CCHmac(operation, saltData!.bytes, saltData!.length, payloadData!.bytes, payloadData!.length, hash!.mutableBytes);
+        return hash?.base64EncodedString
+    }
+
+    private func performEncryptionOperation(operation: Int, key: NSString, iv: NSMutableData, data: NSData) -> NSData? {
+        let keyData: NSData! = (key as NSString).dataUsingEncoding(NSUTF8StringEncoding) as NSData!
+        let keyBytes         = UnsafePointer<Void>(keyData.bytes)
+
+        let dataLength    = data.length
+        let dataBytes     = UnsafePointer<Void>(data.bytes)
+
+        let cryptData    = NSMutableData(length: Int(dataLength) + kCCBlockSizeAES128)
+        var cryptPointer = UnsafeMutablePointer<Void>(cryptData!.mutableBytes)
+        let cryptLength  = cryptData!.length
+
+        let keyLength              = kCCKeySizeAES256
+        let operation: CCOperation = UInt32(operation)
+        // Note 128 is the block length here, not the key length
+        let algoritm:  CCAlgorithm = UInt32(kCCAlgorithmAES128)
+        let options:   CCOptions   = UInt32(kCCModeCBC + kCCOptionPKCS7Padding)
+
+        var numBytesEncrypted: Int = 0
+        var cryptStatus = CCCrypt(operation,
+            algoritm,
+            options,
+            keyBytes, keyLength,
+            iv.bytes,
+            dataBytes, dataLength,
+            cryptPointer, cryptLength,
+            &numBytesEncrypted)
+
+        let success: CCCryptorStatus = Int32(kCCSuccess)
+        if cryptStatus == success {
+            return cryptData!.subdataWithRange(NSRange(location: 0, length: numBytesEncrypted))
+        }
+
+        return nil;
+    }
+
+    private func decodeData(data: NSData) -> (hmac: String, iv: NSMutableData, payload: String) {
+        let decoder = NSKeyedUnarchiver(forReadingWithData: data)
+        let hmac = decoder.decodeObjectForKey(HmacKey) as! String
+        let iv = decoder.decodeObjectForKey(IvKey) as! NSMutableData
+        let payload = decoder.decodeObjectForKey(PayloadKey) as! String
```

Comment:
All these `decodeObjectForKey` calls can fail. Need to go in an `if let` with `as?`. Why not return an optional tuple from `decodeData` so that this function can fail?